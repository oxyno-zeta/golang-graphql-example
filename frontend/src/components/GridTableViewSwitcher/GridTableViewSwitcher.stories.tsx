import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import GridTableViewSwitcher, { Props } from './GridTableViewSwitcher';

export default {
  title: 'Components/GridTableViewSwitcher',
  component: GridTableViewSwitcher,
} as ComponentMeta<typeof GridTableViewSwitcher>;

const Template: ComponentStory<typeof GridTableViewSwitcher> = function C(args: Props) {
  return <GridTableViewSwitcher {...args} />;
};

export const Playground = Template.bind({});
Playground.args = {
  setGridView: () => {},
  gridView: true,
};
