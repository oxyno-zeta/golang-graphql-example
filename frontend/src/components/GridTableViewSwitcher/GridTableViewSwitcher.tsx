import React from 'react';
import { useTranslation } from 'react-i18next';
import Button, { ButtonProps } from '@mui/material/Button';
import ButtonGroup, { ButtonGroupProps } from '@mui/material/ButtonGroup';
import Tooltip from '@mui/material/Tooltip';
import GridViewIcon from '@mui/icons-material/GridView';
import TableRowsIcon from '@mui/icons-material/TableRows';

interface Props {
  setGridView: (input: boolean) => void;
  gridView: boolean;
  buttonGroupProps?: ButtonGroupProps;
  tableButtonProps?: Omit<ButtonProps, 'onClick'>;
  gridButtonProps?: Omit<ButtonProps, 'onClick'>;
}

const defaultProps = {
  buttonGroupProps: {},
  tableButtonProps: {},
  gridButtonProps: {},
};

function GridTableViewSwitcher({ setGridView, gridView, buttonGroupProps, tableButtonProps, gridButtonProps }: Props) {
  // Get translator
  const { t } = useTranslation();

  return (
    <ButtonGroup
      variant="outlined"
      sx={{ border: (theme) => `1px solid ${theme.palette.divider}` }}
      {...buttonGroupProps}
    >
      <Tooltip title={<>{t('common.tableViewTooltip')}</>}>
        <Button
          color="inherit"
          sx={{
            border: 'none',
            padding: '5px 10px',
            minWidth: '46px',
          }}
          onClick={() => {
            setGridView(false);
          }}
          {...tableButtonProps}
        >
          <TableRowsIcon color={gridView ? 'inherit' : 'primary'} />
        </Button>
      </Tooltip>
      <Tooltip title={<>{t('common.gridViewTooltip')}</>}>
        <Button
          color="inherit"
          sx={{ border: 'none', padding: '5px 10px', minWidth: '46px' }}
          onClick={() => {
            setGridView(true);
          }}
          {...gridButtonProps}
        >
          <GridViewIcon color={gridView ? 'primary' : 'inherit'} />
        </Button>
      </Tooltip>
    </ButtonGroup>
  );
}

GridTableViewSwitcher.defaultProps = defaultProps;

export default GridTableViewSwitcher;
