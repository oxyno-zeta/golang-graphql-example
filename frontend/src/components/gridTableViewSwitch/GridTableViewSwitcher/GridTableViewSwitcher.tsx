import React, { useContext } from 'react';
import { useTranslation } from 'react-i18next';
import Button, { ButtonProps } from '@mui/material/Button';
import ButtonGroup, { ButtonGroupProps } from '@mui/material/ButtonGroup';
import Tooltip from '@mui/material/Tooltip';
import SvgIcon from '@mui/material/SvgIcon';
import { mdiViewGridOutline, mdiViewSequential } from '@mdi/js';
import GridTableViewSwitcherContext from '~contexts/GridTableViewSwitcherContext';

export interface Props {
  buttonGroupProps?: ButtonGroupProps;
  tableButtonProps?: Omit<ButtonProps, 'onClick'>;
  gridButtonProps?: Omit<ButtonProps, 'onClick'>;
}

const defaultProps = {
  buttonGroupProps: {},
  tableButtonProps: {},
  gridButtonProps: {},
};

function GridTableViewSwitcher({ buttonGroupProps, tableButtonProps, gridButtonProps }: Props) {
  // Get translator
  const { t } = useTranslation();
  // Get context
  const { isGridViewEnabled, toggleGridTableView } = useContext(GridTableViewSwitcherContext);
  // Get grid view
  const gridView = isGridViewEnabled();

  return (
    <ButtonGroup
      variant="outlined"
      sx={{ border: (theme) => `1px solid ${theme.palette.divider}` }}
      {...buttonGroupProps}
    >
      <Tooltip title={<>{t('common.tableViewTooltip')}</>}>
        <Button
          color="inherit"
          sx={{
            border: 'none',
            padding: '5px 10px',
            minWidth: '46px',
          }}
          onClick={() => {
            // Optimization
            if (gridView) {
              toggleGridTableView();
            }
          }}
          {...tableButtonProps}
        >
          <SvgIcon color={gridView ? 'inherit' : 'primary'}>
            <path d={mdiViewSequential} />
          </SvgIcon>
        </Button>
      </Tooltip>
      <Tooltip title={<>{t('common.gridViewTooltip')}</>}>
        <Button
          color="inherit"
          sx={{ border: 'none', padding: '5px 10px', minWidth: '46px' }}
          onClick={() => {
            // Optimization
            if (!gridView) {
              toggleGridTableView();
            }
          }}
          {...gridButtonProps}
        >
          <SvgIcon color={gridView ? 'primary' : 'inherit'}>
            <path d={mdiViewGridOutline} />
          </SvgIcon>
        </Button>
      </Tooltip>
    </ButtonGroup>
  );
}

GridTableViewSwitcher.defaultProps = defaultProps;

export default GridTableViewSwitcher;
