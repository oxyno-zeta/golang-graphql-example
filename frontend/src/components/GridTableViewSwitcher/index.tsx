import GridTableViewSwitcher from './GridTableViewSwitcher';

export default GridTableViewSwitcher;
