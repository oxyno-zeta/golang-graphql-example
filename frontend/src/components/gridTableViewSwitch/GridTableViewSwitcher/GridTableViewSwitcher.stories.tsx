import React from 'react';
import { type StoryFn, type Meta } from '@storybook/react-vite';
import GridTableViewSwitcher, { type Props } from './GridTableViewSwitcher';
import GridTableViewSwitcherProvider from '../GridTableViewSwitcherProvider';

export default {
  title: 'Components/gridTableViewSwitch/GridTableViewSwitcher',
  component: GridTableViewSwitcher,
} as Meta<typeof GridTableViewSwitcher>;

export const Playground: StoryFn<typeof GridTableViewSwitcher> = function C(args: Props) {
  return (
    <GridTableViewSwitcherProvider>
      <GridTableViewSwitcher {...args} />;
    </GridTableViewSwitcherProvider>
  );
};
