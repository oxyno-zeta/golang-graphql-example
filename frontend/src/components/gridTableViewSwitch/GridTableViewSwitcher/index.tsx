import GridTableViewSwitcher, { Props as GridTableViewSwitcherProps } from './GridTableViewSwitcher';

export default GridTableViewSwitcher;
export type { GridTableViewSwitcherProps };
