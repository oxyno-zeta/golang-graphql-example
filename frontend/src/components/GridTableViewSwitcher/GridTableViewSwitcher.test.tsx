import React from 'react';
import { render, fireEvent } from '@testing-library/react';
// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';

import GridTableViewSwitcher from './GridTableViewSwitcher';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: string) => key }),
}));

describe('GridTableViewSwitcher', () => {
  it('should mark grid in primary color', async () => {
    const onChangeMock = jest.fn();

    const { container, findByRole } = render(<GridTableViewSwitcher onChange={onChangeMock} gridView />);

    expect(container).toMatchSnapshot();
    // Find group
    const groupElement = await findByRole('group');
    expect(groupElement).not.toBeNull();
    expect(groupElement.children).toHaveLength(2);
    expect(groupElement.children[0].firstChild).not.toHaveClass('MuiSvgIcon-colorPrimary');
    expect(groupElement.children[1].firstChild).toHaveClass('MuiSvgIcon-colorPrimary');
    expect(onChangeMock).not.toHaveBeenCalled();
  });

  it('should mark table in primary color', async () => {
    const onChangeMock = jest.fn();

    const { container, findByRole } = render(<GridTableViewSwitcher onChange={onChangeMock} gridView={false} />);

    expect(container).toMatchSnapshot();
    // Find group
    const groupElement = await findByRole('group');
    expect(groupElement).not.toBeNull();
    expect(groupElement.children).toHaveLength(2);
    expect(groupElement.children[0].firstChild).toHaveClass('MuiSvgIcon-colorPrimary');
    expect(groupElement.children[1].firstChild).not.toHaveClass('MuiSvgIcon-colorPrimary');
    expect(onChangeMock).not.toHaveBeenCalled();
  });

  it('should change grid marker when click on grid button and grid view is disabled', async () => {
    const onChangeMock = jest.fn();
    const gridViewValue = false;

    const { container, findByRole } = render(
      <GridTableViewSwitcher onChange={onChangeMock} gridView={gridViewValue} />,
    );

    expect(container).toMatchSnapshot();
    // Find group
    const groupElement = await findByRole('group');
    expect(groupElement).not.toBeNull();
    expect(groupElement.children).toHaveLength(2);
    expect(groupElement.children[0].firstChild).toHaveClass('MuiSvgIcon-colorPrimary');
    expect(groupElement.children[1].firstChild).not.toHaveClass('MuiSvgIcon-colorPrimary');

    fireEvent.click(groupElement.children[1], 'click');

    expect(onChangeMock).toHaveBeenCalled();
    expect(onChangeMock).toHaveBeenCalledWith(true);
  });

  it('should change grid marker when click on table button and grid view is enabled', async () => {
    const onChangeMock = jest.fn();
    const gridViewValue = true;

    const { container, findByRole } = render(
      <GridTableViewSwitcher onChange={onChangeMock} gridView={gridViewValue} />,
    );

    expect(container).toMatchSnapshot();
    // Find group
    const groupElement = await findByRole('group');
    expect(groupElement).not.toBeNull();
    expect(groupElement.children).toHaveLength(2);
    expect(groupElement.children[0].firstChild).not.toHaveClass('MuiSvgIcon-colorPrimary');
    expect(groupElement.children[1].firstChild).toHaveClass('MuiSvgIcon-colorPrimary');

    fireEvent.click(groupElement.children[0], 'click');

    expect(onChangeMock).toHaveBeenCalled();
    expect(onChangeMock).toHaveBeenCalledWith(false);
  });

  it("shouldn't change grid marker when click on grid button and grid view is already enabled", async () => {
    const onChangeMock = jest.fn();
    const gridViewValue = true;

    const { container, findByRole } = render(
      <GridTableViewSwitcher onChange={onChangeMock} gridView={gridViewValue} />,
    );

    expect(container).toMatchSnapshot();
    // Find group
    const groupElement = await findByRole('group');
    expect(groupElement).not.toBeNull();
    expect(groupElement.children).toHaveLength(2);
    expect(groupElement.children[0].firstChild).not.toHaveClass('MuiSvgIcon-colorPrimary');
    expect(groupElement.children[1].firstChild).toHaveClass('MuiSvgIcon-colorPrimary');

    fireEvent.click(groupElement.children[1], 'click');

    expect(onChangeMock).not.toHaveBeenCalled();
  });

  it("shouldn't change grid marker when click on table button and grid view is already disabled", async () => {
    const onChangeMock = jest.fn();
    const gridViewValue = false;

    const { container, findByRole } = render(
      <GridTableViewSwitcher onChange={onChangeMock} gridView={gridViewValue} />,
    );

    expect(container).toMatchSnapshot();
    // Find group
    const groupElement = await findByRole('group');
    expect(groupElement).not.toBeNull();
    expect(groupElement.children).toHaveLength(2);
    expect(groupElement.children[0].firstChild).toHaveClass('MuiSvgIcon-colorPrimary');
    expect(groupElement.children[1].firstChild).not.toHaveClass('MuiSvgIcon-colorPrimary');

    fireEvent.click(groupElement.children[0], 'click');

    expect(onChangeMock).not.toHaveBeenCalled();
  });
});
