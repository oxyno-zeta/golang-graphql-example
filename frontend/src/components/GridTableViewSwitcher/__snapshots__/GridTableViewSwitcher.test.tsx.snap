// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridTableViewSwitcher should change grid marker when click on grid button and grid view is disabled 1`] = `
<div>
  <div
    class="MuiButtonGroup-root MuiButtonGroup-outlined css-kuz68f-MuiButtonGroup-root"
    role="group"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorInherit MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorInherit css-1268z6w-MuiButtonBase-root-MuiButton-root"
      data-mui-internal-clone-element="true"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 5H21V9H3V5M3 10H21V14H3V10M3 15H21V19H3V15Z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorInherit MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorInherit css-1268z6w-MuiButtonBase-root-MuiButton-root"
      data-mui-internal-clone-element="true"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 11H11V3H3M5 5H9V9H5M13 21H21V13H13M15 15H19V19H15M3 21H11V13H3M5 15H9V19H5M13 3V11H21V3M19 9H15V5H19Z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`GridTableViewSwitcher should change grid marker when click on table button and grid view is enabled 1`] = `
<div>
  <div
    class="MuiButtonGroup-root MuiButtonGroup-outlined css-kuz68f-MuiButtonGroup-root"
    role="group"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorInherit MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorInherit css-1268z6w-MuiButtonBase-root-MuiButton-root"
      data-mui-internal-clone-element="true"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 5H21V9H3V5M3 10H21V14H3V10M3 15H21V19H3V15Z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorInherit MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorInherit css-1268z6w-MuiButtonBase-root-MuiButton-root"
      data-mui-internal-clone-element="true"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 11H11V3H3M5 5H9V9H5M13 21H21V13H13M15 15H19V19H15M3 21H11V13H3M5 15H9V19H5M13 3V11H21V3M19 9H15V5H19Z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`GridTableViewSwitcher should mark grid in primary color 1`] = `
<div>
  <div
    class="MuiButtonGroup-root MuiButtonGroup-outlined css-kuz68f-MuiButtonGroup-root"
    role="group"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorInherit MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorInherit css-1268z6w-MuiButtonBase-root-MuiButton-root"
      data-mui-internal-clone-element="true"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 5H21V9H3V5M3 10H21V14H3V10M3 15H21V19H3V15Z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorInherit MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorInherit css-1268z6w-MuiButtonBase-root-MuiButton-root"
      data-mui-internal-clone-element="true"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 11H11V3H3M5 5H9V9H5M13 21H21V13H13M15 15H19V19H15M3 21H11V13H3M5 15H9V19H5M13 3V11H21V3M19 9H15V5H19Z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`GridTableViewSwitcher should mark table in primary color 1`] = `
<div>
  <div
    class="MuiButtonGroup-root MuiButtonGroup-outlined css-kuz68f-MuiButtonGroup-root"
    role="group"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorInherit MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorInherit css-1268z6w-MuiButtonBase-root-MuiButton-root"
      data-mui-internal-clone-element="true"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 5H21V9H3V5M3 10H21V14H3V10M3 15H21V19H3V15Z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorInherit MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorInherit css-1268z6w-MuiButtonBase-root-MuiButton-root"
      data-mui-internal-clone-element="true"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 11H11V3H3M5 5H9V9H5M13 21H21V13H13M15 15H19V19H15M3 21H11V13H3M5 15H9V19H5M13 3V11H21V3M19 9H15V5H19Z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`GridTableViewSwitcher shouldn't change grid marker when click on grid button and grid view is already enabled 1`] = `
<div>
  <div
    class="MuiButtonGroup-root MuiButtonGroup-outlined css-kuz68f-MuiButtonGroup-root"
    role="group"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorInherit MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorInherit css-1268z6w-MuiButtonBase-root-MuiButton-root"
      data-mui-internal-clone-element="true"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 5H21V9H3V5M3 10H21V14H3V10M3 15H21V19H3V15Z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorInherit MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorInherit css-1268z6w-MuiButtonBase-root-MuiButton-root"
      data-mui-internal-clone-element="true"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 11H11V3H3M5 5H9V9H5M13 21H21V13H13M15 15H19V19H15M3 21H11V13H3M5 15H9V19H5M13 3V11H21V3M19 9H15V5H19Z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`GridTableViewSwitcher shouldn't change grid marker when click on table button and grid view is already disabled 1`] = `
<div>
  <div
    class="MuiButtonGroup-root MuiButtonGroup-outlined css-kuz68f-MuiButtonGroup-root"
    role="group"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorInherit MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorInherit css-1268z6w-MuiButtonBase-root-MuiButton-root"
      data-mui-internal-clone-element="true"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 5H21V9H3V5M3 10H21V14H3V10M3 15H21V19H3V15Z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorInherit MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorInherit css-1268z6w-MuiButtonBase-root-MuiButton-root"
      data-mui-internal-clone-element="true"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 11H11V3H3M5 5H9V9H5M13 21H21V13H13M15 15H19V19H15M3 21H11V13H3M5 15H9V19H5M13 3V11H21V3M19 9H15V5H19Z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
